### ISSUE TITLE
TypeError: t.target is undefined

### ISSUE THREAD

**Issue by sojusnik** [ActivityWatch/activitywatch#611]
- [x] I am on the [latest](https://github.com/ActivityWatch/activitywatch/releases/latest) ActivityWatch version.
 - [x] I have searched the issues of this repo and believe that this is not a duplicate.

**OS name and version**: Ubuntu 20.10
**ActivityWatch version**: v0.11.0

## Describe the bug
After updating to v0.11.0, I get the following error message, when changing the `duration default value` under http://127.0.0.1:5600/#/settings: `TypeError: t.target is undefined. See dev console (F12) and/or server logs for more info.`

Here's the relevant part of the log:

```
2021-06-16 17:04:53 [INFO ]: Using storage method: peewee  (aw_server.main:32)
2021-06-16 17:04:53 [INFO ]: Starting up...  (aw_server.main:37)
2021-06-16 17:04:53 [INFO ]: Using database file: /home/sojusnik/.local/share/activitywatch/aw-server/peewee-sqlite.v2.db  (aw_datastore.storages.peewee:116)
2021-06-16 17:04:53 [INFO ]:  * Running on http://localhost:5600/ (Press CTRL+C to quit)  (werkzeug:113)
2021-06-16 17:05:09 [INFO ]: Received heartbeat after pulse window, inserting as new event. (bucket: aw-watcher-afk_mir)  (aw_server.api:262)
2021-06-16 17:05:36 [INFO ]: 404 (127.0.0.1): [33mGET /img/icons/apple-touch-icon-152x152.png HTTP/1.1[0m  (flask:25)
2021-06-16 17:06:03 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:06:03 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:06:03 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:06:03 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:06:03 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:08:19 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:08:19 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:08:19 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:08:19 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:08:19 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:09:09 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:09:09 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:09:09 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:09:09 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:09:09 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:21:59 [INFO ]: 404 (127.0.0.1): [33mGET /img/icons/apple-touch-icon-152x152.png HTTP/1.1[0m  (flask:25)
2021-06-16 17:22:01 [INFO ]: 308 (127.0.0.1): [32mGET /api HTTP/1.1[0m  (flask:25)
2021-06-16 17:23:28 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:23:28 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:23:28 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:23:28 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:23:28 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:24:15 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:24:34 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:24:42 [WARNING]: Gap was of negative duration but could be safely merged (-3222.408s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:24:50 [WARNING]: Gap was of negative duration but could be safely merged (-3507.265s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:25:33 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:25:33 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:25:33 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:25:33 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:25:33 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:11 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-511.164s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-185.183s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-1037.105s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-195.22s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-4173.679s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-15374.187s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-280.543s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-250.3s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-0.001s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration and could NOT be safely merged (-606.488s). This warning will only show once per batch.  (aw_transform.flood:54)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-190.213s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-411.043s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-3507.265s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-295.56s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-305.553s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:17 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:21 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:22 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:24 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:25 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:32:28 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-3507.265s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-295.56s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-3222.408s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-315.46s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-300.66s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-887.087s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-305.553s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:36:40 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:38 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:38 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:38 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:38 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:38 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:58 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:58 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:58 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:58 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:37:58 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:16 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:16 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:16 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:16 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:16 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:57 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:57 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:57 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:57 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:38:57 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:39:16 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:39:16 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:39:16 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:39:16 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:39:16 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:40:37 [INFO ]: 404 (127.0.0.1): [33mGET /img/icons/apple-touch-icon-152x152.png HTTP/1.1[0m  (flask:25)
2021-06-16 17:41:00 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:00 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:00 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:00 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:00 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:23 [ERROR]: Exception on /api/0/query/ [POST]  (aw-server:1891)
Traceback (most recent call last):
  File "flask/app.py", line 1950, in full_dispatch_request
  File "flask/app.py", line 1936, in dispatch_request
  File "flask_restx/api.py", line 375, in wrapper
  File "flask/views.py", line 89, in view
  File "flask_restx/resource.py", line 44, in dispatch_request
  File "aw_server/rest.py", line 280, in post
  File "aw_server/api.py", line 290, in query2
  File "iso8601/iso8601.py", line 195, in parse_date
iso8601.iso8601.ParseError: Unable to parse date string 'Invalid date'
2021-06-16 17:41:23 [INFO ]: 500 (127.0.0.1): [35m[1mPOST /api/0/query/ HTTP/1.1[0m  (flask:25)
2021-06-16 17:41:24 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:24 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:24 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:24 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:41:24 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:03 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:03 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:03 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:03 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:03 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-887.087s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-305.553s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:25 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:37 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:47 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:58 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:58 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:58 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:58 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:42:58 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-305.553s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:10 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:30 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:37 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:48 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:55 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:56 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:56 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:56 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:56 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:44:56 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:45:06 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:02 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:02 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:02 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:02 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:02 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:04 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:16 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:27 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:36 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:46 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:53 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:53 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:53 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:53 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:49:53 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:04 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:14 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:22 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:22 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:22 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:22 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 17:53:22 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:34 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:44 [WARNING]: Gap was of negative duration but could be safely merged (-5.003s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:51 [WARNING]: Gap was of negative duration but could be safely merged (-48489.32s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:51 [WARNING]: Gap was of negative duration but could be safely merged (-9158.248s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:51 [WARNING]: Gap was of negative duration but could be safely merged (-2526.657s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:51 [WARNING]: Gap was of negative duration but could be safely merged (-335.78s). This message will only show once per batch.  (aw_transform.flood:46)
2021-06-16 18:03:51 [WARNING]: Gap was of negative duration but could be safely merged (-365.592s). This message will only show once per batch.  (aw_transform.flood:46)
```

**Comment by kulbhushanchand** [comment]
I'm having the **same/similar** issue. Someone in the [forum mentioned the same](https://forum.activitywatch.net/t/how-to-update-to-v11/1274)

![image](https://user-images.githubusercontent.com/17742733/122260582-13d37f00-cef1-11eb-9350-f8899c0a3faf.png)

**Comment by ErikBjare** [comment]
Thanks for reporting!

Fixed in https://github.com/ActivityWatch/aw-webui/commit/41a5f579fe22fb2f3450604187a59d42a26d95bf, hoping to get a patch release out soon.

**Comment by kulbhushanchand** [comment]
> **OS name and version**: Windows 10 x64
> **ActivityWatch version**: v0.11.0
> 
> I'm having the **same/similar** issue. Someone in the [forum mentioned the same](https://forum.activitywatch.net/t/how-to-update-to-v11/1274)
> 
> ![image](https://user-images.githubusercontent.com/17742733/122260582-13d37f00-cef1-11eb-9350-f8899c0a3faf.png)

@ErikBjare Looks like my above comment is more related to this [forum-post](https://forum.activitywatch.net/t/how-to-update-to-v11/1274/3). Should I post a separate issue for that? or the already posted forum post is enough.

**Comment by ErikBjare** [comment]
@kulbhushanchand I think that is actually the same problem as described in this issue (and should be fixed by the linked commit). The forum post looks like something different (note it mentions 'label', not 'target' or 'value').

**Comment by kulbhushanchand** [comment]
Oh, you are right. Later I did a clean install and removed all the previous data (including log files, which I have already exported at a safe place). Now, this is what I am seeing. 

```
TypeError: Cannot read property 'label' of undefined. See dev console (F12) and/or server logs for more info.
```

![image](https://user-images.githubusercontent.com/17742733/122355193-563ba100-cf6f-11eb-9078-f8c9f3d384cd.png)

![image](https://user-images.githubusercontent.com/17742733/122355384-78cdba00-cf6f-11eb-9693-364a21f84bb1.png)

**Comment by ErikBjare** [comment]
@kulbhushanchand Did you try resetting the categories to the defaults? (you can export your existing ones before you do)

The 'missing c' in logs is a good hint and leads me to https://github.com/ActivityWatch/aw-webui/blob/889a911e584145569f82d207e564f9686b735cb2/src/util/datasets.js#L34. It's undoubtedly the cause of the error (as the branch arm would result in `undefined` passing through the map), but I'm not sure under which conditions the `if(c)` ends up being false.

Perhaps a workaround would be to simply filter out the `undefined`'s that slip through? Unsure about what that would mean in practice though, and would be nice to know the underlying cause...

**Comment by kulbhushanchand** [comment]
Yes, I did reset the categories to their defaults before, but the problem was there. So to be sure, just now, I did the same again and here are my findings.

- Reset the categories to their defaults and the `label` error stops.
- Manually add a new category from any of your previous categories, and there is a chance `label` error shows up, along with `value` error.
- If I tries to manually import my categories from previously exported json file, the `label` error definitely returns.

**Comment by kulbhushanchand** [comment]
Yes, I did reset the categories to their defaults before, but the problem was there. So to be sure, just now, I did the same again and here are my findings.

- Reset the categories to their defaults and the `label` error stops.
- Manually add a new category from any of your previous categories, and there is a chance `label` error shows up, along with `value` error.
- If I tries to manually import my categories from previously exported json file, the `label` error definitely returns.
